### PR TITLE
OIDC Integration

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Server-side implementation of the [WebID-OIDC (OpenID Connect) authentication spec](https://github.com/solid/webid-oidc-spec).

Major features:

1) Replaces WebID-TLS authentication with WebID-OIDC. WebID-TLS auth will continue to be available on a maintenance `v3.5` branch of solid-server.
2) Moves tests into one of two directories, `/test/unit` and `/test/integration`, corresponding.
3) Improves post-install "out of the box" user experience -- sets up server front page in multi-user mode, installs default signup & signin apps, etc.
4) Increases unit and integration test coverage for the server.

### CODE REVIEW

File: lib/identity-provider.js [hunk 1]
```diff
@@ -524,15 +531,83 @@ IdentityProvider.prototype.getGraph = function (uri, callback) {
     })
 }
 
-// Handle POST requests on account creation
-IdentityProvider.prototype.post = function (req, res, next) {
-  if (!req.body) {
+/**
+<<<<<<< 6477d67eec3b591c4dd226307ff634c1a26e40c6
```
Comment (nicola) [inline lib/identity-provider.js]: whoops?

Comment (dmitrizagidulin) [inline lib/identity-provider.js]: Doh!! Thanks :)

**Comment by bshamblen** [comment]
@dmitrizagidulin Please let me know if there are any pieces of the OIDC project you could you a hand implementing. I'd be happy to lend a hand, if you need it.

**Comment by dmitrizagidulin** [comment]
@bshamblen sure thing! Stop by our channel on Gitter - https://gitter.com/solid/chat and we can arrange a quick call to sync up

**Comment by dmitrizagidulin** [comment]
And by gitter.com I mean gitter.im, sigh

**Comment by dmitrizagidulin** [comment]
oidc integration for solid-server PR ready for review. /cc @timbl @deiu @dan-f

**Comment by RubenVerborgh** [comment]
Has been merged into the v4.0.0 release branch (#572).

**Comment by nicola** [comment]
do we have OIDC integrated now?

**Comment by RubenVerborgh** [comment]
Yes, in https://github.com/solid/node-solid-server/tree/release/v4.0.0

**Pull request closed without merging**